Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -1070,28 +1070,50 @@ JSC_DEFINE_JIT_OPERATION(operationWasmArrayNew, EncodedJSValue, (Instance* insta
     ASSERT(arraySignature.is<ArrayType>());
     Wasm::FieldType fieldType = arraySignature.as<ArrayType>()->elementType();
 
+    size_t elementSize = fieldType.type.elementSize();
+
     JSWebAssemblyArray* array = nullptr;
-    switch (fieldType.type.kind) {
-    case Wasm::TypeKind::I32:
-    case Wasm::TypeKind::F32: {
+
+    if (elementSize == sizeof(uint8_t)) {
+        FixedVector<uint8_t> values(size);
+        // `value` must be an unboxed int (since the typechecker guarantees that its type is i32);
+        // so it's safe to truncate it to 8 bits
+        encValue = encValue & UCHAR_MAX;
```

Comment:
Might be better to use the max type constants from https://en.cppreference.com/w/cpp/types/integer that match the fixed width types.

Also this might be better as a `switch` statement on the size if possible.